Sources_v3/Controllers/UserController/UserController_Tests.swift

```diff
@@ -393,6 +393,160 @@ final class UserController_Tests: StressTestCase {
         // Assert updater is called with correct `userId`
         XCTAssertEqual(env.userUpdater!.unmuteUser_userId, controller.userId)
     }
+    
+    // MARK: - Flag user
+    
+    func test_flagUser_propogatesError() {
+        // Simulate `flag` call and catch the completion.
+        var completionError: Error?
+        controller.flag { [callbackQueueID] in
+            AssertTestQueue(withId: callbackQueueID)
+            completionError = $0
+        }
+        
+        // Simulate network response with the error.
+        let networkError = TestError()
+        env.userUpdater!.flagUser_completion!(networkError)
+        
+        // Assert error is propogated.
+        AssertAsync.willBeEqual(completionError as? TestError, networkError)
+    }
+    
+    func test_flagUser_propogatesNilError() {
+        // Simulate `flag` call and catch the completion.
+        var completionIsCalled = false
+        controller.flag { [callbackQueueID] error in
+            // Assert callback queue is correct.
+            AssertTestQueue(withId: callbackQueueID)
+            // Assert there is no error.
+            XCTAssertNil(error)
+            completionIsCalled = true
+        }
+        
+        // Simulate successful network response.
+        env.userUpdater!.flagUser_completion!(nil)
+        
+        // Assert completion is called.
+        AssertAsync.willBeTrue(completionIsCalled)
+    }
+    
+    func test_flagUser_callsUserUpdater_withCorrectValues() {
+        // Simulate `flag` call.
+        controller.flag()
+        
+        // Assert updater is called with correct `flag`
+        XCTAssertEqual(env.userUpdater!.flagUser_flag, true)
+        // Assert updater is called with correct `userId`
+        XCTAssertEqual(env.userUpdater!.flagUser_userId, controller.userId)
+    }
+    
+    func test_flagUser_keepsControllerAlive() {
```

Comment:
😊

Reply:
@evsaev I forgot one thing - do you think we could make this test simpler? Remove all unnecessary asserts and things we already test elsewhere? Keep just code related to testing it keeps it alive.

Reply:
fixed in a700a03db59c86ecd1fe241925124e009f3035c3